src/list_reader.c

```diff
@@ -66,41 +73,43 @@ void ListIterator_Free(struct indexIterator *self) {
 
 static size_t LR_NumEstimated(void *ctx) {
   ListIterator *lr = ctx;
-  return lr->size;
+  return VecSimQueryResult_Len(lr->list);
 }
 
 static size_t LR_LastDocId(void *ctx) {
   ListIterator *lr = ctx;
-  return lr->list[lr->size - 1].id;
+  return lr->lastDocId;
 }
 
 static size_t LR_NumDocs(void *ctx) {
   ListIterator *lr = ctx;
-  return lr->size;
+  return VecSimQueryResult_Len(lr->list);
 }
 
 static void LR_Abort(void *ctx) {
   ListIterator *lr = ctx;
-  lr->offset = lr->size;
+  setEof(lr, 1);
 }
 
 static void LR_Rewind(void *ctx) {
   ListIterator *lr = ctx;
-  lr->offset = 0;
+  VecSimQueryResult_IteratorFree(lr->iter);
+  lr->iter = VecSimQueryResult_List_GetIterator(lr->list);
+  lr->lastDocId = 0;
+  setEof(lr, 0);
```

Comment:
Even if it's empty?

Reply:
We will figure it out the first time we run.
This check isn't done on other iterators.